src/palimpzest/prompts/prompt_factory.py

```diff
@@ -0,0 +1,721 @@
+    COT_MOA_PROPOSER_JOB_INSTRUCTION,
+)
+from palimpzest.prompts.util_phrases import (
+    ONE_TO_MANY_OUTPUT_FORMAT_INSTRUCTION,
+    ONE_TO_ONE_OUTPUT_FORMAT_INSTRUCTION,
+)
+
+
+class PromptFactory:
+    """Factory class for generating prompts for the Generator given the input(s)."""
+    BASE_SYSTEM_PROMPT_MAP = {
+        PromptStrategy.COT_BOOL: COT_BOOL_BASE_SYSTEM_PROMPT,
+        PromptStrategy.COT_BOOL_IMAGE: COT_BOOL_BASE_SYSTEM_PROMPT,
+        PromptStrategy.COT_QA: COT_QA_BASE_SYSTEM_PROMPT,
+        PromptStrategy.COT_QA_CRITIC: None,
+        PromptStrategy.COT_QA_REFINE: None,
+        PromptStrategy.COT_QA_IMAGE: COT_QA_BASE_SYSTEM_PROMPT,
+        PromptStrategy.COT_QA_IMAGE_CRITIC: None,
+        PromptStrategy.COT_QA_IMAGE_REFINE: None,
+        PromptStrategy.COT_MOA_PROPOSER: COT_MOA_PROPOSER_BASE_SYSTEM_PROMPT,
+        PromptStrategy.COT_MOA_AGG: COT_MOA_AGG_BASE_SYSTEM_PROMPT,
+    }
+    BASE_USER_PROMPT_MAP = {
+        PromptStrategy.COT_BOOL: COT_BOOL_BASE_USER_PROMPT,
+        PromptStrategy.COT_BOOL_IMAGE: COT_BOOL_BASE_USER_PROMPT,
+        PromptStrategy.COT_QA: COT_QA_BASE_USER_PROMPT,
+        PromptStrategy.COT_QA_CRITIC: BASE_CRITIQUE_PROMPT,
+        PromptStrategy.COT_QA_REFINE: BASE_REFINEMENT_PROMPT,
+        PromptStrategy.COT_QA_IMAGE: COT_QA_BASE_USER_PROMPT,
+        PromptStrategy.COT_QA_IMAGE_CRITIC: BASE_CRITIQUE_PROMPT,
+        PromptStrategy.COT_QA_IMAGE_REFINE: BASE_REFINEMENT_PROMPT,
+        PromptStrategy.COT_MOA_PROPOSER: COT_MOA_PROPOSER_BASE_USER_PROMPT,
+        PromptStrategy.COT_MOA_AGG: COT_MOA_AGG_BASE_USER_PROMPT,
+    }
+
+    def __init__(self, prompt_strategy: PromptStrategy, model: Model, cardinality: Cardinality) -> None:
+        self.prompt_strategy = prompt_strategy
+        self.model = model
+        self.cardinality = cardinality
+
+    def _get_context(self, candidate: DataRecord, input_fields: list[str]) -> str:
+        """
+        Returns the context for the prompt.
+
+        Args:
+            candidate (DataRecord): The input record.
+            input_fields (list[str]): The input fields.
+
+        Returns:
+            str: The context.
+        """
+        # get context from input record (project_cols will be None if not provided in kwargs)
+        context: dict = candidate.to_dict(include_bytes=False, project_cols=input_fields)
+
+        # TODO: this does not work for image prompts
+        # TODO: this ignores the size of the `orignal_messages` in critique and refine prompts
+        # cut down on context based on window length
+        if self.model in [Model.LLAMA3, Model.MIXTRAL]:
+            total_context_len = len(json.dumps(context, indent=2))
+
+            # sort fields by length and progressively strip from the longest field until it is short enough;
+            # NOTE: 6000 is a rough estimate which leaves room for the rest of the prompt text
+            while total_context_len * TOKENS_PER_CHARACTER > 6000:
```

Comment:
nit: Can you make this 6000 a constant (or at least a variable)?

Reply:
Sure thing!